docs/docs/mdx/programmatically-creating-pages.md

```diff
@@ -0,0 +1,369 @@
+
+## Source MDX pages from the filesystem
+
+To let Gatsby know that you'll be working with MDX content you need to
+add `gatsby-mdx` to the plugins array in your `gatsby-config.js`
+file.
+
+You'll need to use `gatsby-source-filesystem` and tell it to source
+"posts" from a folder called `content/posts` located in the project's
+root.
+
+> _NOTE: `gatsby-mdx` uses `.mdx` by default as a file extension to
+> recognize which files to use. You can also [use `.md` as a file
+> extension](api-reference/options/extensions) if you want._
+
+```javascript
+// gatsby-config.js
+module.exports = {
+  plugins: [
+    // Add support for *.mdx files in gatsby
+    "gatsby-mdx",
+
+    // Add a collection called "posts" that looks
+    // for files in content/posts/
+    {
+      resolve: "gatsby-source-filesystem",
+      options: {
+        name: "posts",
+        path: `${__dirname}/content/posts/`,
+      },
+    },
+  ],
+}
+```
+
+You can read about
+[gatsby-source-filesystem](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-filesystem#readme)
+if you'd like to learn more.
+
+## Add MDX Files
+
+Before you can write any GraphQL queries, and programmatically create
+pages, you need to add some content.
+
+Make a folder called `content/posts` and create two files in it called
+`blog-1.mdx` and `blog-2.mdx`. You can do this on the command line in
+a terminal by using the following commands from the root of your
+project.
+
+```sh
+mkdir -p content/posts
+touch content/posts/blog-{1,2}.mdx
+```
+
+> _NOTE:
+> `mkdir -p path/to/a/directory` will create every folder in the path if
+> it does not exist._
+>
+> _`touch <filename>` will create an empty file named `<filename>`. The
+> brackets (`{}`) are [an
+> expansion](https://twitter.com/kentcdodds/status/1083399683932868609?s=19)
+> which means we can create multiple files in one command._
+
+Open up each of the files you just created and add some content.
+
+```md
+---
+title: "Blog Post 1"
+---
+
+Trying out MDX
+```
+
+```md
+---
+title: "Blog Post 2"
+---
+
+Gatsby is the best
+```
+
+## Generate slugs
+
+Since MDX posts are being sourced outside of `src/pages`, each post
+needs to be given a slug which tells Gatsby the URL to render to.
+
+If you want to set the URLs in your frontmatter, you can skip this step.
+
+```js
+// gatsby-node.js
+const { createFilePath } = require("gatsby-source-filesystem")
+
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+
+  // We only want to operate on `Mdx` nodes. If we had content from a
+  // remote CMS we could also check to see if the parent node was a
+  // `File` node here
+  if (node.internal.type === "Mdx") {
+    const value = createFilePath({ node, getNode })
+
+    createNodeField({
+      // Name of the field you are adding
+      name: "slug",
+      // Individual MDX node
+      node,
+      // Generated value based on filepath with "blog" prefix
+      value: `/blog${value}`,
```

Comment:
No separating `/` needed?

Reply:
It isn't needed. I'll update the comment above to elaborate on the why.